impl/src/main/kotlin/nl/nuts/consent/bridge/diagnostics/CordaFlowHealthIndicators.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Nuts consent bridge
+ * Copyright (C) 2019 Nuts community
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package nl.nuts.consent.bridge.diagnostics
+
+import nl.nuts.consent.bridge.SchedulerProperties
+import nl.nuts.consent.bridge.rpc.CordaService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.scheduling.annotation.Async
+import org.springframework.scheduling.annotation.EnableAsync
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import kotlin.math.floor
+
+
+/**
+ * Shared class for logic calling Corda Flows
+ */
+abstract class FlowHealthIndicator : HealthIndicator {
+    var lastCheck : CordaService.PingResult = CordaService.PingResult(true)
+
+    @Autowired
+    protected lateinit var schedulerProperties: SchedulerProperties
+
+    @Autowired
+    protected lateinit var cordaService: CordaService
+
+    override fun health(): Health {
+        var now = System.currentTimeMillis()
```

Comment:
I'd use the Java 8 Date/Time API here, something like this:
```
val lastCheck = LocalDateTime.now()
val age = Duration.between(lastCheck, LocalDateTime.now())
if (age > Duration.ofSeconds(delay * 2)) {
 print age.toMinutes()
 etc...
 ...
}
```

Maybe you can also use Java 8's Date/Time classes for the configuration (Duration instead of long for initial delay), but I'm not sure if this Spring version supports that.